pkg/github/check_api.go

```diff
@@ -3,68 +3,73 @@ package github
 import (
 	"context"
 	"fmt"
+	"strings"
 	"time"
 
-	"github.com/google/go-github/v39/github"
+	"github.com/google/go-github/v43/github"
 	"github.com/rs/zerolog/log"
+	"github.com/shurcooL/githubv4"
+	"github.com/terraform-tools/terraform-checker/pkg/terraform"
 )
 
-func (t *CheckEvent) UpdateCheckRun(cr GhCheckRun, status bool, message string) {
-	var conclusion string
-	if status {
-		conclusion = "success"
-	} else {
-		conclusion = "failure"
+func (e *CheckEvent) CreateCheckRun(check terraform.TfCheck) (GhCheckRun, error) {
+	checkRunName := fmt.Sprintf("terraform-check %v %v", check.Name(), check.RelDir())
+	log.Info().Msgf("Create check run %s on repo %s PR %s", checkRunName, e.GetRepo().GetFullName(), e.GetPRUrl())
+
+	externalId := fmt.Sprintf("%s:%s", check.RelDir(), check.Name())
```

Comment:
should probably be moved to a function `encodeExternalId` and `decodeExternalId` in the future
so we can keep encode/decode close to one another
could be nice to also start with a version like `1:<dir>:<check_name>` that way we can have `2:<dir>:<check_name>:<additional_field>`

Reply:
done for the encode/decode functions.
we'll see later for improvements